src/lib.rs

```diff
@@ -527,6 +505,7 @@ fn run_scripts(ctx: &Ctx, template_composite: &TemplateComposite) -> Result<()>
 mod tests {
     use super::*;
     pub use crate::cli_opt::*;
+    use crate::ctx::tests::new_ctx_from;
```

Comment:
IMHO, It's better for tests if sub-modules used stuff from a parent (for test content of sub/side modules are black-box). So move the definition of `new_ctx_from` into this mod.